frontend/package.json

```diff
@@ -32,60 +32,65 @@
   },
   "homepage": "https://github.com/cryptag/leapchat#readme",
   "dependencies": {
-    "@babel/preset-env": "^7.20.2",
-    "@emoji-mart/data": "^1.1.2",
     "atob": "^2.1.2",
-    "babel-loader": "^9.1.2",
+    "babel-loader": "^7.1.5",
     "btoa": "^1.2.1",
-    "crypto-browserify": "^3.12.0",
-    "emoji-datasource-apple": "^14.0.0",
-    "emoji-js": "^3.7.0",
-    "emoji-mart": "^5.5.2",
+    "copy-webpack-plugin": "^4.6.0",
+    "emoji-datasource-apple": "^3.0.0",
+    "emoji-js": "^3.5.0",
+    "emoji-mart": "^1.0.1",
+    "extract-text-webpack-plugin": "^3.0.2",
     "guid": "0.0.12",
-    "jquery": "^3.6.3",
-    "js-sha512": "^0.8.0",
-    "jsdom": "^21.1.0",
-    "markdown-it": "^13.0.1",
-    "react": "^18.2.0",
-    "react-bootstrap": "^2.7.1",
-    "react-dom": "^18.2.0",
-    "react-icons": "^4.7.1",
-    "react-redux": "^8.0.5",
-    "redux": "^4.2.1",
-    "stream-browserify": "^3.0.0",
-    "utf8": "^3.0.0"
+    "jquery": "^3.5.1",
+    "js-sha512": "^0.3.0",
+    "markdown-it": "^8.4.2",
+    "prop-types": "^15.7.2",
+    "react": "^15.7.0",
+    "react-bootstrap": "^0.31.5",
+    "react-dom": "^15.7.0",
+    "react-icons": "^2.2.7",
+    "react-redux": "^5.1.2",
+    "redux": "^3.7.2",
+    "utf8": "^2.1.2"
   },
   "devDependencies": {
-    "@babel/core": "^7.20.12",
-    "@babel/eslint-parser": "^7.19.1",
-    "@babel/eslint-plugin": "^7.19.1",
-    "@babel/plugin-proposal-object-rest-spread": "^7.20.7",
-    "@babel/preset-react": "^7.18.6",
-    "@babel/register": "^7.18.9",
-    "@webpack-cli/generators": "^3.0.1",
-    "babel-plugin-system-import-transformer": "^4.0.0",
+    "babel-eslint": "^7.2.3",
+    "babel-plugin-system-import-transformer": "^3.1.0",
     "babel-plugin-transform-class-properties": "^6.24.1",
     "babel-plugin-transform-object-rest-spread": "^6.26.0",
+    "babel-preset-latest": "^6.24.1",
+    "babel-preset-react": "^6.24.1",
+    "babelify": "^7.3.0",
```

Comment:
Just so I'm clear: we do not want to upgrade the packages in the frontend the latest just yet?

The point of the `frontend` directory was to get the packages on the latest and get a passing build that way. It looks like you reverted some packages in the root dir and got it passing that way.

If we don't want to upgrade now, I'll revert the `frontend` directory to the way I had it and put it on a feature branch for later use.

Reply:
I'm sorry, I thought your goal was to dockerize a working version of the app ASAP, so yeah I reverted the deps.

> Just so I'm clear: we do not want to upgrade the packages in the frontend the latest just yet?

That's what I'm thinking, yes. But if we run into issues adding features then we'll have no choice and need to update the deps and let's revisit this. Let's try to add features first!

We can immediately use the docker work you did here and that I just tweaked :slightly_smiling_face: .

> I'll revert the `frontend` directory to the way I had it and put it on a feature branch for later use.

That sounds perfect, thank you :pray: .